### DISCUSSION TITLE
Modify a TakeProfit

### DISCUSSION THREAD

**Discussion by dragonflyuk** [hootnot/oanda-api-v20#201] [Q&A] [answered]
I've been playing with the code, in a very basic test python script, however I'm struggling to modify the takeprofit on an existing position?

Do you have to replace the existing order? In which case is the new order automatically linked as a take profit?

**Comment by hootnot** [comment]
Since you didn't show any code I assume that you used https://oanda-api-v20.readthedocs.io/en/latest/contrib/support/takeprofitdetails.html to get a position (this example a MarketOrder with a TP).
That position will have a TP order and changing that order should keep it linked to the position.

See also: https://oanda-api-v20.readthedocs.io/en/latest/endpoints/orders/orderreplace.html

**Comment by dragonflyuk** [comment, marked as answer]
Thanks, I didn't include because at that point I didn't have have any :-)

However I've been playing, and for the benefit of others, Ive got as far as :-

```
r = orders.OrderReplace(
    accountID,
    td["takeProfitOrder"]["id"],
    data={
        "order": {
            "units": -int(td["currentUnits"]),
            "instrument": "EUR_USD",
            "price": "{0:.5f}".format(stp),
            "type": "TAKE_PROFIT",
            "tradeID": td["id"],
        }
    },
)

```

**Reply by hootnot** [reply]
I've applied `black` to your snippet for readability. You can experiment with code on the OANDA test environment.
